.github/workflows/go-tests-other-os.yml

```diff
@@ -7,15 +7,17 @@ on:
       - .github/workflows/go-tests-other-os.yml
       - .github/actions/**
       - codeql-workspace.yml
+env:
+  GO_VERSION: '1.21.0'
```

Comment:
Do we want to automatically get new patch versions when they're released? Something like `1.21.x` or `~1.21.0` might work.